src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt

```diff
@@ -0,0 +1,185 @@
+package uk.gov.dluhc.printapi.service
+
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import org.mockito.ArgumentCaptor
+import org.mockito.Captor
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.any
+import org.mockito.kotlin.capture
+import org.mockito.kotlin.given
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import uk.gov.dluhc.printapi.database.entity.PrintDetails
+import uk.gov.dluhc.printapi.database.entity.PrintRequestStatus
+import uk.gov.dluhc.printapi.database.entity.Status
+import uk.gov.dluhc.printapi.database.entity.Status.DISPATCHED
+import uk.gov.dluhc.printapi.database.entity.Status.PENDING_ASSIGNMENT_TO_BATCH
+import uk.gov.dluhc.printapi.database.entity.Status.RECEIVED_BY_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.database.entity.Status.SENT_TO_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.database.repository.PrintDetailsRepository
+import uk.gov.dluhc.printapi.printprovider.models.BatchResponse.Status.FAILED
+import uk.gov.dluhc.printapi.printprovider.models.BatchResponse.Status.SUCCESS
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidRequestId
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildPrintDetails
+import uk.gov.dluhc.printapi.testsupport.testdata.model.buildBatchResponse
+import uk.gov.dluhc.printapi.testsupport.testdata.model.buildProcessPrintResponseMessage
+import java.time.Clock
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.Status as ResponseStatus
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.StatusStep as ResponseStatusStep
+
+@ExtendWith(MockitoExtension::class)
+class PrintResponseProcessingServiceTest {
+    private lateinit var service: PrintResponseProcessingService
+
+    @Mock
+    private lateinit var printDetailsRepository: PrintDetailsRepository
+
+    @Mock
+    private lateinit var idFactory: IdFactory
+
+    private val fixedClock = Clock.fixed(Instant.now().minusSeconds(10000), ZoneOffset.UTC)
+
+    @Captor
+    private lateinit var captor: ArgumentCaptor<List<PrintDetails>>
+
+    private val now = OffsetDateTime.now(fixedClock)
+
+    @BeforeEach
+    fun setup() {
+        service = PrintResponseProcessingService(printDetailsRepository, idFactory, fixedClock)
+    }
+
+    @Nested
+    inner class ProcessBatchResponses {
+        @Test
+        fun `should update print details`() {
```

Comment:
How about `should update all print requests within each batch`? on the basis that we are processing some batches, and the expected side effect is an update to the print requests ?

(Also, I'm not fussed whether you choose to use the term "print request" or "print details" in the method name - the business and the Print Provider refer to it as a "print request", but our modelling in dynamo has used the term "print details" .... not sure what the new name will be when we move over to JPA? @Neil-Massey ?)

Reply:
Thanks, for the suggestion @NathanRussellValtech ,
I will rename it accordingly. I think we will be using print requests when we move over to JPA, so happy to rename it to print request.

Reply:
That is a fair comment; completely agree 👍 💯